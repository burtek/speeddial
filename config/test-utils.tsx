import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { Provider as StoreProvider } from 'react-redux';

import { store } from '@@data/index';
import { ThemedApp } from 'src/theme/themed-app';

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider store={store}>
        <ThemedApp>
            {children}
        </ThemedApp>
    </StoreProvider>
);

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
