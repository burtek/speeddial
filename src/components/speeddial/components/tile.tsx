import type { Theme } from '@mui/material';
import { Card as MuiCard, styled } from '@mui/material';
import type { CardTypeMap } from '@mui/material/Card';
import type { OverridableComponent } from '@mui/material/OverridableComponent';
import type { CSSObject } from '@mui/material/styles';
import type { MUIStyledCommonProps } from '@mui/system';

import { TILE_WIDTH } from './_constants';


interface TileProps {
    isDragging: boolean;
    transform?: string;
    transition?: string;
}

const styles = ({ isDragging, transform, transition }: TileProps & MUIStyledCommonProps<Theme>): CSSObject => ({
    cursor: isDragging ? 'grabbing' : 'pointer',
    pointerEvents: isDragging ? 'none' : undefined,
    textDecoration: 'none',
    transform,
    transition,
    width: TILE_WIDTH,
    zIndex: isDragging ? 1 : undefined
});

// `styled` disallows `component` overriding :(
// see https://mui.com/material-ui/guides/typescript/#complications-with-the-component-prop
export const Tile = styled<OverridableComponent<CardTypeMap<{ component?: 'a' | 'div' }, 'a' | 'div'>>>(
    MuiCard,
    { shouldForwardProp: prop => !['theme', 'isDragging', 'transform', 'transition'].includes(prop) }
)<TileProps>(styles);
