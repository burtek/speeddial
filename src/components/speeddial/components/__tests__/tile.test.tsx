import { render } from '@@config/test-utils';

import { Tile } from '../tile';


describe('sppeddial/componnets/tile', () => {
    describe.each([
        { type: 'tile', component: 'div' as const },
        { type: 'achor tile', component: 'a' as const }
    ])('$type', ({ component }) => {
        it.each([
            { isDragging: false },
            { isDragging: true }
        ])('should match snapshot with isDragging=$isDragging', ({ isDragging }) => {
            const { container } = render(<Tile component={component} isDragging={isDragging}><div>Some content</div></Tile>);

            expect(container).toMatchSnapshot();
        });
    });
});
