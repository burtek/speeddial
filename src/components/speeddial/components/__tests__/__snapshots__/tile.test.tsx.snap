// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sppeddial/componnets/tile > 'achor tile' > should match snapshot with isDragging=false 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 150px;
}

<div>
  <a
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div>
      Some content
    </div>
  </a>
</div>
`;

exports[`sppeddial/componnets/tile > 'achor tile' > should match snapshot with isDragging=true 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  cursor: -webkit-grabbing;
  cursor: grabbing;
  pointer-events: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 150px;
  z-index: 1;
}

<div>
  <a
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div>
      Some content
    </div>
  </a>
</div>
`;

exports[`sppeddial/componnets/tile > 'tile' > should match snapshot with isDragging=false 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 150px;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div>
      Some content
    </div>
  </div>
</div>
`;

exports[`sppeddial/componnets/tile > 'tile' > should match snapshot with isDragging=true 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  cursor: -webkit-grabbing;
  cursor: grabbing;
  pointer-events: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 150px;
  z-index: 1;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div>
      Some content
    </div>
  </div>
</div>
`;
